### PR TITLE
Rebase Nix flake PR

### PR DESCRIPTION
**Pull request by j-baker**
@koiuo this is the same as your PR, but with CI and a dev-shell setup. I also moved openssl into the buildInputs section of the dependencies, as it is a runtime dependency.

### CODE REVIEW

**Review by mara-schulke — APPROVED** [review]
LGTM except the env file

File: .envrc [hunk 1]
```diff
@@ -0,0 +1 @@
+use flake
```
[the remark this answers is not in the document]
Comment (j-baker) [reply .envrc:1]: I’m not sure what you mean. Envrc files are specific to direnv users?

[the remark this answers is not in the document]
Comment (mara-schulke) [reply .envrc:1]: Im not sure, but this looks like it should not be committed (imo) as it is relevant to you, but not to most people

**Comment by koiuo** [comment]
@j-baker re: runtime dependency

I believe rustls depends on openssl (can’t doublecheck from my phone) and cargo build fails when rustc reaches rustls if OpenSSL headers not available (at least on NixOS that’s the case, rustls’s cargo build script calls pkgconfig).

I’m not sure how crane works internally, but I’d conclude from this behavior that openssl is a build dependency.

File: .envrc (hunk 1 above)
[the remark this answers is not in the document]
Comment (mara-schulke) [reply .envrc]: I removed it for now as there was no further discussion on this. If you absolutely need it feel free to reopen / open a new mr 🙈

**Pull request merged**